cmd/kube-scheduler/app/options/deprecated.go

```diff
@@ -120,20 +122,19 @@ func (o *DeprecatedOptions) ApplyTo(cfg *kubeschedulerconfig.KubeSchedulerConfig
 		}
 	}
 
-	// The following deprecated options affect the only existing profile that is
-	// added by default.
+	// Deprecated flags have an effect iff no config file was provided, in which
+	// case this function expects a default KubeSchedulerConfiguration instance,
+	// which has a single profile.
 	profile := &cfg.Profiles[0]
 	if len(o.SchedulerName) > 0 {
 		profile.SchedulerName = o.SchedulerName
 	}
-	if o.HardPodAffinitySymmetricWeight != interpodaffinity.DefaultHardPodAffinityWeight {
```

Comment:
This now always adds an InterPodAffinityArgs object, even when the HardPodAffinityWeight value was unchanged from the default.

Does that mean if other defaulted fields are added to InterPodAffinityArgs, `getPluginArgsOrDefault ` will use the object we define here and will not apply v1alpha2.InterPodAffinityArgs defaults for those other fields?

Should this do a decode of a v1alpha2.InterPodAffinityArgs to pick up any other defined defaults for that type, then override the HardPodAffinityWeight field?

Reply:
Good point. Obtaining the default object and overriding the field is probably the right solution. However, I don't expect InterPodAffinityArgs to change in the near future (read that as "until component config is GA" and so we can remove the deprecated flags).

Reply:
Added comment in `SetDefaults_InterPodAffinityArgs`